### PR TITLE
Enable `consistent-return` rule

### PR DESCRIPTION
**Pull request by omh1280**
Add [`consistent-return`](http://eslint.org/docs/rules/consistenet-return) ESLint rule. As mentioned in #5456, many functions rely on returning either a `value` or `undefined`. 

I suppressed nearly all of the errors inline, with notable exceptions being in `KmlDataSource.js` and `gulpfile.js`.

### CODE REVIEW

**Comment by pjcozzi** [comment]
Hmm, I dunno, this one seems troublesome to have to workaround.  Perhaps just keep the real fixes here and do not enable this.

Up to @mramato

**Comment by mramato** [comment]
I agree that all of this error suppression is bad.  What I was expecting here is that if a function sometime returns a value and sometimes doesn't, this rule would just make us specifically state `return undefined;` in those cases.  This makes it more clear that we really want `undefined` instead of it being a bug.

For example, we have a bunch of code like this:

```javascript
if(!foo) {
    return;
}
//Do stuff
return result;
```

That would get turned into this.

```javascript
if(!foo) {
    return undefined;
}
//Do stuff
return result;
```

@omh1280 am I misunderstanding the rule, or did you just take the approach of suppressing instead of adding the explicit `undefined`?  I think if you remove all of the suppression and do what I suggest above, this rule may be worth it.

Did this reveal any real problems?  If not, then I realize this may be bordering on stylistic choices, so I'm okay with just punting on this rule completely if @pjcozzi thinks it's not worth it. (If it did reveal real bugs, then it may be worth keeping).

**Comment by omh1280** [comment]
You're totally right. I misunderstood the rule.

Updated all the suppression to instead follow the rule.

File: gulpfile.js [hunk 1]
```diff
@@ -143,11 +143,12 @@ gulp.task('cloc', ['build'], function() {
         child_process.exec(cmdLine, function(error, stdout, stderr) {
             if (error) {
                 console.log(stderr);
-                return reject(error);
+                reject(error);
```
Comment (omh1280) [inline gulpfile.js]: Are these changes correct? I don't see the difference between calling `reject` and returning it

**Comment by omh1280** [comment]
@mramato can you prioritize this? The glob change means that the eslint tests aren't running on travis

**Comment by mramato** [comment]
After looking at the updated code, I'm not sure this rule is worth it and since it didn't find any bugs, it's unlikely that we would introduce problems in the future because of adding this rule.  Thanks for evaluating this @omh1280!

As discussed offline, open another PR for the glob changes.

**Comment by pjcozzi** [comment]
Sounds good!

**Pull request closed without merging**